### ISSUE TITLE
Old Storybook Style Story Tree

### ISSUE THREAD

**Issue by webOS101** [storybookjs/storybook#2863]
### Issue details

_Please provide issue details here - What did you expect to happen? What happened instead?_ 
In the newest versions of Storybook, clicking on a category no longer opens up the first story associated with that category.  We used to have a category (`storiesOf()`) and each category had only a single entry with an empty (`' '`) title.  This worked fairly well for our use case.  Is it possible to reproduce this style?  Or, is it possible to have a root story node (empty category)?

### Steps to reproduce

_Please provide necessary steps for reproduction of this issue. Describe the exact steps a maintainer has to take to make the problem occur. If the problem is non-trivial to reproduce, please link a repository or provide some code snippets._

_(A screencast can be useful for visual bugs, but it is not a substitute for a textual description.)_

Create a story with an empty title on Storybook 2 and compare with Storybook 3.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 3.3.10

### Affected platforms 

- _If UI related, please indicate browser, OS, and version_
- _If dependency related, please include relevant version numbers_
- _If developer tooling related, please include the platform information_

### Screenshots / Screencast / Code Snippets (Optional)

```js
storiesOf('About', module)
	.add(
		' ',
		withInfo('A Tour of Sampler')(() => (
...
```

**Comment by Hypnosphi** [comment]
CC: @igor-dv

**Comment by igor-dv** [comment]
I think we can add support for it, but need to think how to make it less ugly and in the line with existing API.

**Comment by Hypnosphi** [comment]
`import {add} from '@storybook/react'`?

**Comment by igor-dv** [comment]
@Hypnosphi I think it looks more like a new api... I would like somehow to use an existing one... Maybe to be able to do something like 

```js
storiesOf(module)
	.add('Story' () => {});
```
But I don't like it either =(

**Comment by PinkaminaDianePie** [comment]
This bug makes navigation twice as hard. Each time 2 clicks instead of one :(

**Comment by Grsmto** [comment]
Coming from Docz, that would be really welcome!

It's quite obvious that this is a big UX issue both from dev and user experience when you see that you have to do this to display a homepage:
![Screenshot 2019-04-16 at 17 45 44](https://user-images.githubusercontent.com/527559/56242865-97725680-606f-11e9-8af6-3880aaecaba3.jpg)

There seem to be a couple of issue referencing this one so I suppose this is a need from the community.
What would be the effort to implement this? Or maybe there is a higher plan for menu/navigation customisation that we are not aware of?

**Comment by shilman** [comment]
cc @domyen

**Comment by domyen** [comment]
**TLDR**: This behavior is a better fit for the upcoming [Storybook Docs](https://medium.com/storybookjs/storybook-docs-sneak-peak-5be78445094a) feature not Storybook.

There are two common SB use cases:
1. Isolated UI Component development
2. Component library documentation

Currently, Storybook is intended for 1. isolated component development – helping devs create components. The more folks get into 2. component library documentation, the less ideal Storybook is (for now!). This issue is an example where the use cases are blurred thus causing a split on what's the best user experience.

If you're using SB for 1, the current treeview behavior is ideal. It's easy to predict what's going to happen if you click on a collapsed list item – you will expand it and reveal it's contents. This is identical behavior as the treeview in VSCode, OSX Finder, Windows Explorer etc. suffice to say it's a pretty universal pattern. I think we should stick with it.

If you're using SB for 2, the current treeview behavior results in an extra click. I remember the previous behavior is SB2.0 and def appreciate that it can be annoying for this use case. I think the old behavior is ideal for a docs site, where clicking on a list item, reveals different "sections" inside that list item. 

For example: clicking "The Vue Instance" reveals a bunch of sections of the doc and effectively shows you the first section without an extra click. <a href="https://cl.ly/ae1072382e00" target="_blank"><img src="https://d3crtwyc2nw0fl.cloudfront.net/items/2G2M0w3Y1z0H3O2X031W/Screen%20Recording%202019-04-19%20at%2004.57%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>

**The good news** is that we're building a new documentation-mode where we should definitely consider the click-to-auto-view-first-story treeview behavior (perhaps as an option?). This would help folks expecting a sidebar nav similar to Docz or Styleguidist.

**Comment by doublejosh** [comment]
Supposedly this was possible by adding the desired story to the stories array first.
https://stackoverflow.com/questions/53724555/specify-default-page-for-storybook

However, that did not work for me. It might be due to some stories lacking a root, etc.

**Comment by stephiescastle** [comment]
> **The good news** is that we're building a new documentation-mode where we should definitely consider the click-to-auto-view-first-story treeview behavior (perhaps as an option?). This would help folks expecting a sidebar nav similar to Docz or Styleguidist.

@domyen Is this still being considered? This is still a UX issue for our team.

**Comment by shilman** [comment]
@stephiescastle The original request to auto-navigate to the first story in the component has been fixed already (in 5.x). We're also planning to overhaul Storybook Docs and the future behavior could be closer to what you're looking for. That said, we haven't figured that out yet, nor do we have a specific timeline for those improvements yet.

Any feedback on what specifically you'd like to see would be much appreciated! 🙏

**Comment by stephiescastle** [comment]
@shilman Thanks for the update! I should have elaborated. Clicking on a component does auto-navigate to its first story, but what we'd like is to have the first item within a `data-nodetype="group"` to be active when expanding that group. This is to avoid two clicks to view something within that group. Should I open up a new issue?
